### PR TITLE
AK: Introduce an `as<OutputType>()` cast 

### PR DESCRIPTION
**Pull request by tcl3**
There are many places in the codebase where we currently use this pattern and variations of it:

```c++
if (is<Bar>(foo)) {
    auto& bar = verify_cast<Bar>(foo);
    do_stuff(bar);
}
```

The `as<OutputType>(input)` cast returns a pointer to an object of type `OutputType`, which is null if `input` can't be cast to `OutputType`.

This allows the above to be rewritten as:

```c++
if (auto* bar = as<Bar>(foo)) {
    do_stuff(*bar);
}
```

This change also replaces the `as` function used in the JSSpecCompiler with this new AK function, as both of them existing at the same time was causing a compile error.

I've updated `LibWeb/Dump.cpp` to use `as` casts as part of this PR to show how the new cast is used. I'd be happy to apply a similar change in more places if people think this is a good approach.

### CODE REVIEW

**Comment by mattco98** [comment]
Maybe the function should be called `cast` to have some symmetry with `verify_cast`? Just a suggestion though, feel free to ignore.

File: Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Compiler/Passes/CFGSimplificationPass.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include "Compiler/Passes/CFGSimplificationPass.h"
 #include "AST/AST.h"
 #include "Function.h"
+#include <AK/TypeCasts.h>
```
Comment (DanShaders) [inline Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Compiler/Passes/CFGSimplificationPass.cpp:10]: The "conventional" order of includes in JSSC is libraries, blank line, JSSC stuff.

**Review by DanShaders — COMMENTED** [review]
JSSC changes looks fine but I wonder if we can avoid them altogether with `template<typename T> RefPtr<T> as(RefPtr<T>)` overload defined in `AK/RefPtr.h`.

(Oh my god, last JSSC PR was 2.5 months ago, I should really continue working on it)

**Comment by awesomekling** [comment]
`as<T>` feels like it could make a better/shorter name for `verify_cast<T>` to me.
Maybe `as_if<T>` for the version that returns null if it's not a `T`. :yakshed:

File: Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Compiler/Passes/CFGSimplificationPass.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/Compiler/Passes/CFGSimplificationPass.cpp:10]: Hmm. That's not a convention we use anywhere else.

**Pull request closed without merging**